### DISCUSSION TITLE
Help ESP8266 GET Request

### DISCUSSION THREAD

**Discussion by 1andre19** [micropython/micropython#13020] [ESP8266]
I have this code that I tested on Wokwi, an online IoT and embedded systems emulator. This code works fine on Wokwi in an ESP32, however when I tried to load this code to an actual ESP8266 it doesn't work.

**This is my code:**

```
import network
import urequests
import time

ssid = 'Wokwi-GUEST'
password = ''

station = network.WLAN(network.STA_IF)

station.active(True)
station.connect(ssid, password)

while not station.isconnected():
    time.sleep(1)

print('Connected to WiFi')

url = 'https://apex.oracle.com/pls/apex/a01285600/linkpost/insert?tipo=TEST&valor=20'
headers = {
    "User-Agent": "Mozilla/5.0 (Windows NT 6.1; WOW64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/56.0.2924.76 Safari/537.36",
    "Upgrade-Insecure-Requests": "1",
    "DNT": "1",
    "Accept": "text/html,application/xhtml+xml,application/xml;q=0.9,/;q=0.8",
    "Accept-Language": "en-US,en;q=0.5",
    "Accept-Encoding": "gzip, deflate"
}

while True:
    if station.isconnected():
        try:
            response = urequests.get(url, headers=headers)
            print(response.status_code)
            print(response.text)
            response.close()
        except Exception as e:
            print('Error on HTTP request:', e)
    else:
        print('WiFi not connected')

    time.sleep(10)  
```
    
 **This is the error:**
    
Connected to WiFi
Error on HTTP request: -40

Many thanks in advance.

**Comment by skyguy** [comment]
This may be because of the large certificate list which that server sends back as part of the TLS handshake. I remember reading somewhere that the ESP8266 only has a small TCP buffer of 2000 bytes, but the server you're connecting to sends back a 2917-byte TLS chunk.

The 8266 probably can't decode the handshake to make a succesful connection. You may have to use a device with a bit more memory to cope with TLS.
